atlas_splitter/barrel_splitter/somatosensory_barrels.py

```diff
@@ -0,0 +1,294 @@
+        the resulting binary mask after performing the logical AND operation (bool values).
+    """
+    mask = positions_to_mask(positions, annotation).astype(bool)
+    region_mask = np.isin(
+        annotation.raw,
+        indices,
+    ).astype(bool)
+    layer_barrel = np.logical_and(region_mask, mask).astype(bool)
+
+    return layer_barrel
+
+
+def add_hierarchy_child(
+    parent: Dict[str, Any], id_: int, name: str, acronym: str
+) -> Dict[str, Any]:
+    """
+    Add a new child to a hierarchical structure.
+
+    Args:
+        parent: The parent structure to which the child is being added.
+        id: The unique identifier for the child.
+        name: The name of the child.
+        acronym: The acronym for the child.
+
+    Returns:
+        A new copy of the parent structure with the child added to it.
+    """
+    new_child = copy.deepcopy(parent)
+    new_child["parent_structure_id"] = parent["id"]
+    new_child["acronym"] = acronym
+    new_child["name"] = name
+    new_child["id"] = id_
+    new_child["st_level"] = parent["st_level"] + 1
+    new_child["graph_order"] = parent["graph_order"] + 1
+    return new_child
+
+
+def edit_hierarchy(  # pylint: disable=too-many-arguments
+    hierarchy: HierarchyDict,
+    new_ids: Dict[str, Dict[str, int]],
+    region_map: RegionMap,
+    start_index: int,
+    children_names: List[str],
+    layers: List[str],
+) -> None:
+    """Edit in place the hierarchy to include new children volumes of a given
+    region. Implemented to integrated the barrel columns into [SSp-bfd]
+    Barrel cortex in Primary Somatosensory cortex.
+
+    Note:
+        The following attributes of the created nodes are copies of the
+        parent attributes (see see http://api.brain-map.org/doc/Structure.html for some
+        definitions):
+        - atlas_id
+        - color_hex_triplet
+        - hemisphere_id (always set to 3 for the AIBS Mouse P56)
+        - graph_order (the structure order in a graph flattened via in order traversal)
+        - ontology_id (always set to 1 for the AIBS Mouse P56)
+        - st_level
+        No proper value of graph_order can be set for a new child. This is why it is left
+        unchanged.
+
+    Args:
+        hierarchy (HierarchyDict): brain regions hierarchy dict
+        new_ids (Dict[int, Dict[str, int]]): set of new ids
+        start_index (int): index of the starting brain region (Isocortex)
+        children_names (list): list of children volumes to be integrated
+        layers (list): list of layers to be integrated
+    """
+    # pylint: disable=too-many-locals
+    hierarchy_levels = np.array(region_map.get(start_index, attr="acronym", with_ascendants=True))
+    iso_index = np.where(hierarchy_levels == "Isocortex")[0][0]
+    hierarchy_levels = hierarchy_levels[:iso_index]
+    hierarchy_ = get_isocortex_hierarchy(hierarchy)
+
+    for acronym in hierarchy_levels[::-1]:
+        hierarchy_ = get_hierarchy_by_acronym(hierarchy_, acronym)
```

Comment:
I made some changes, but I think it would simplify the code if the "SSp-bfd" was hard coded in `edit_hierarchy`, not passed from `split_barrels`. Let me know

Reply:
whatever you think is best